hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/SimpleForeignKeyDescriptor.java

```diff
@@ -285,32 +288,25 @@ public EntityMappingType findContainingEntityMapping() {
 		throw new UnsupportedOperationException();
 	}
 
-	@Override
-	public String getReferringTableExpression() {
-		return keyColumnContainingTable;
-	}
-
 	@Override
 	public void visitReferringColumns(ColumnConsumer consumer) {
 		consumer.accept( keyColumnContainingTable, keyColumnExpression, jdbcMapping );
 	}
 
-	@Override
-	public String getTargetTableExpression() {
-		return targetColumnContainingTable;
-	}
-
 	@Override
 	public void visitTargetColumns(ColumnConsumer consumer) {
 		consumer.accept( targetColumnContainingTable, targetColumnExpression, jdbcMapping );
 	}
 
 	@Override
-	public boolean areTargetColumnNamesEqualsTo(String[] columnNames) {
-		if ( columnNames.length != 1 ) {
-			return false;
+	public AssociationKey getAssociationKey() {
+		if ( associationKey == null ) {
+			List<String> associationKeyColumns = new ArrayList<>( 1 );
+			associationKeyColumns.add( keyColumnExpression );
+
+			associationKey = new AssociationKey( keyColumnContainingTable, associationKeyColumns );
```

Comment:
Do we allow for the possibility that `associationKeyColumns` is mutable later? Otherwise `Collections.singletonList()` would be more appropriate and less confusing.